From 7a842785a5059e69c6688ebfe4e16c1171f62406 Mon Sep 17 00:00:00 2001
From: "Du, Changbin" <changbinx.du@intel.com>
Date: Wed, 5 Nov 2014 13:04:54 +0800
Subject: [PATCH] usb: phy-intel-cht: set proper fsm status on USB_EVENT_VBUS
 event

On USB_EVENT_VBUS event, set b_sess_vld if id floated (device mode),
else set a_vbus_vld if id grounded (host mode).

Change-Id: I2d5f51b45531f7cdfafb91eabcae54afff4d0385
Signed-off-by: Du, Changbin <changbinx.du@intel.com>
---
 drivers/usb/phy/phy-intel-cht.c |    5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/usb/phy/phy-intel-cht.c b/drivers/usb/phy/phy-intel-cht.c
index 64f4d3d..bd03edd 100644
--- a/drivers/usb/phy/phy-intel-cht.c
+++ b/drivers/usb/phy/phy-intel-cht.c
@@ -215,7 +215,10 @@ static int cht_otg_handle_notification(struct notifier_block *nb,
 	/* USB_EVENT_VBUS: vbus valid event */
 	case USB_EVENT_VBUS:
 		dev_info(cht_otg_dev->phy.dev, "USB_EVENT_VBUS vbus valid\n");
-		cht_otg_dev->fsm.b_sess_vld = 1;
+		if (cht_otg_dev->fsm.id)
+			cht_otg_dev->fsm.b_sess_vld = 1;
+		else
+			cht_otg_dev->fsm.a_vbus_vld = 1;
 		schedule_work(&cht_otg_dev->fsm_work);
 		state = NOTIFY_OK;
 		break;
-- 
1.7.9.5

